src/fides/api/models/privacy_experience.py

```diff
@@ -97,38 +255,131 @@ def create(
+        upsert_privacy_experiences_after_config_update(db, experience_config, regions)
+        # Link Privacy Notices to this Privacy Experience config via the ExperienceNotices table
+        link_notices_to_experience_config(
+            db, notice_ids=privacy_notice_ids, experience_config=experience_config
+        )
+
         return experience_config
 
     def update(self, db: Session, *, data: dict[str, Any]) -> PrivacyExperienceConfig:
         """
-        Overrides the base update method to automatically bump the version of the
-        PrivacyExperienceConfig record and also create a new PrivacyExperienceConfigHistory entry
+        Updates a PrivacyExperienceConfig and related resources which is a large exercise!
+
+        - Updates the PrivacyExperienceConfig details
+        - For each supplied translation, add, update, or delete translations so they match
+        the translations in the update request
+        - For each remaining translation, if the translation or the config has changed,
+        create a historical record for auditing purposes
+        - Link or unlink regions (via Privacy Experiences)
+        - Link or unlink Privacy Notices
         """
-        resource, updated = update_if_modified(self, db=db, data=data)
+        request_translations = data.pop("translations", [])
+        regions = data.pop("regions", [])
+        privacy_notice_ids = data.pop("privacy_notice_ids", [])
+
+        # Do a patch update of the existing privacy experience config if applicable
+        config_updated = update_if_modified(self, db=db, data=data)
+
+        for translation_data in request_translations:
+            existing_translation: Optional[
+                ExperienceTranslation
+            ] = self.get_translation_by_language(db, translation_data.get("language"))
+            if existing_translation:
+                # Do a patch update of the existing experience translation if applicable
+                translation_updated: bool = update_if_modified(
+                    existing_translation,
+                    db=db,
+                    data={**translation_data, "experience_config_id": self.id},
+                )
+                translation = existing_translation
+            else:
+                translation_updated = True
+                # Create a new Experience Translation since one doesn't exist
+                translation = ExperienceTranslation.create(
+                    db,
+                    data={**translation_data, "experience_config_id": self.id},
+                )
 
-        if updated:
-            history_data = create_historical_data_from_record(resource)
-            history_data["experience_config_id"] = self.id
+            if config_updated or translation_updated:
+                existing_version: float = translation.version or 0.0
+                history_data: dict = create_historical_data_from_record(self)
```

Comment:
Correct! When speccing out original requirements, I wondered if we both wanted a historical record of both "privacy notices" and "regions" available.  

But I [checked with product](https://ethyca.atlassian.net/wiki/spaces/PM/pages/2905997363/Multi-language+and+experience+config+implementation+questions+answers#My-first-batch-of-backend-questions-for-consent-multitranslation%3A) and  it sounded like it is fine to leave this off, because it doesn't add additional value when preferences are saved, because we have the "region" the user responded from and the "privacy notice" against which the user is consenting.